### ISSUE TITLE
[BUG]: First Time Loading the App Users Get Connection Not Configured Error

### ISSUE THREAD

**Issue by MDRooks** [microsoft/powerapps-tools#143]
When Opening the app on a mobile device for the first time Users receive the error message:
"connection not configured for this data source"
This error is resolved when closing and re-opening the app and does not re-occur.

1. Open the app on a mobile device for the fist time.

Error:
connection not configured for this data source

All connection and data sources are configured and working fine except for the very first load of the app. We should not get an error the first time we open it.

Only able to reproduce this one time for each user. Have seen it on tablets, android phones, and IPhones.

**Comment by mr-dang-msft** [comment]
Hi @MDRooks ,
It is expected that the app will produce this error until you have connected to your own SharePoint lists. 

You will need to remove and add your own SharePoint connections to the app. Then you need to save and publish the app for its changes to take effect. If you had done the publishing, it does take a refresh for the published version to appear.

**Comment by MDRooks** [comment]
@mr-dang-msft thanks for the response!

I've updated the  all of the SharePoint connections to all point to the SharePoint site on my tenant and refreshed them (just in case). I've also republished the app multiple times since then, but am still seeing the same error with new users.

It seems as if the first time users are logging in the app is loading up faster than the connection is completed setting up and giving an error because it never re-occurs after the first time they open the app. (Also only occurs on mobile devices).

**Comment by davecawthorn** [comment]
Same problem, trying to deploy in the org now and not getting past testing with a small number of users

**Comment by mr-dang-msft** [comment]
@MDRooks ,
Do users receive the error each time you republish a new version?

**Comment by MDRooks** [comment]
@mr-dang-msft No, this only occurs the first time our users open the app and then never re-occurs.

**Comment by davecawthorn** [comment]
@mr-dang-msft We've shared the app using 'everyone in the organisation', should that make a difference?  Also, the users we were testing with are refreshing the app, clearing the cache and even signing in / out of PA all to no avail.  Our SP site is a team site set to public as per the guide and all the connectors are pointing at it.  We're doing more testing later, will update.

**Comment by mr-dang-msft** [comment]
Can you submit a support ticket on this topic and gather a session id in the instance that the issue happens?

Session Id: aka.ms/pa/sessionid
Support ticket: aka.ms/ppac

**Comment by davecawthorn** [comment]
@mr-dang-msft thanks Brian, that's been raised as support ticket 120032322004069

**Comment by MDRooks** [comment]
I have a ticket raised as well. Was able to reproduce the error while on a call with support.
Ticket: #‎19197244‎

**Comment by MDRooks** [comment]
Unfortunately Microsoft was not able to help with this issue. However, I was able to solve it myself through trial/error.

I removed the O365 Outlook and Office365UserProfile connections in the app and then re-added them. For some reason this resolved the issue and users no longer are receiving the error the first time they open the app in mobile.

I've sent the fix to Microsoft and hopefully they can provide some more insight as to why this would resolve the error and why it occurred in the first place.

Thanks so much for the quick response and looking into this for me @mr-dang-msft!
